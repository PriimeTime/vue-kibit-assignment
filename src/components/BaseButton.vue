<script setup lang="ts">
import { computed } from "vue";

const props = defineProps<{
  flat?: boolean;
}>();

const emit = defineEmits(["click"]);

const buttonClass = computed(() => {
  return props.flat
    ? "p-2 min-w-20 text-teal-900 flex justify-center items-center whitespace-nowrap"
    : "p-2 min-w-20 rounded-md text-white border-teal-900 border-2 bg-teal-900 hover:text-teal-900 hover:bg-white flex justify-center items-center whitespace-nowrap";
});
</script>

<template>
  <button @click="emit('click')" :class="buttonClass">
    <slot />
  </button>
</template>
