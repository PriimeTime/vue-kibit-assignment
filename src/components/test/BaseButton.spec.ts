import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import BaseButton from "@/components/BaseButton.vue";

describe("BaseButton", () => {
  it("applies correct classes when 'flat' prop is true", () => {
    const wrapper = mount(BaseButton, {
      props: { flat: true },
    });
    expect(wrapper.attributes("class")).toContain("text-teal-900");
    expect(wrapper.attributes("class")).not.toContain("rounded-md");
  });

  it("emits 'click' event when clicked", async () => {
    const wrapper = mount(BaseButton);
    await wrapper.trigger("click");
    expect(wrapper.emitted("click")).toBeTruthy();
  });

  it("matches snapshot", () => {
    const wrapper = mount(BaseButton, {
      props: { flat: false },
      slots: {
        default: "Click me",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
