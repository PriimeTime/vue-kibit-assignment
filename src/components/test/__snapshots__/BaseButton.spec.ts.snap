// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseButton > matches snapshot 1`] = `"<button class="p-2 min-w-20 rounded-md text-white border-teal-900 border-2 bg-teal-900 hover:text-teal-900 hover:bg-white flex justify-center items-center whitespace-nowrap">Click me</button>"`;
